File: Geometry/Poins_on_a_Line/points_on_a_line.cpp
#include <bits/stdc++.h>

int main()
{
  int n = 0;
  std::cin >> n;

  std::set<int> x, y;
  int x_i = 0, y_i = 0;
  bool same_line = true;

  for (int i = 0; i < n; ++i)
  {
    std::cin >> x_i >> y_i;

    x.insert(x_i);
    y.insert(y_i);
  }

  if ((x.size() == 1 || y.size() == 1))
    std::cout << "YES" << std::endl;
  else
    std::cout << "NO" << std::endl;

  return 0;
}

File: Geometry/Poins_on_a_Line/points_on_a_line.py
#!/bin/python3


if __name__=="__main__":
    n = int(input())

    x_0, y_0 = map(int, input().split())

    same_line = True
    for _ in range(n - 1):
        x_i, y_i = map(int, input().split())

        if not(x_i == x_0 or y_i == y_0):
            same_line = False

    print("YES" if same_line else "NO")
